lib/MySQL_HostGroups_Manager.cpp

```diff
@@ -1953,6 +1958,58 @@ bool MySQL_HostGroups_Manager::commit(
 		pthread_mutex_unlock(&GloVars.checksum_mutex);
 	}
 
+	// fill Hostgroup_Manager_Mapping with latest records
+	if (hgsm_mysql_servers_checksum != table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS] ||
+		hgsm_mysql_replication_hostgroups_checksum != table_resultset_checksum[HGM_TABLES::MYSQL_REPLICATION_HOSTGROUPS])
+	{
+		proxy_info("Checksum for table 'mysql_servers': old:0x%lX new:0x%lX\n", hgsm_mysql_servers_checksum, table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS]);
+		proxy_info("Checksum for table 'mysql_replication_hostgroups': old:0x%lX new:0x%lX\n", hgsm_mysql_replication_hostgroups_checksum, table_resultset_checksum[HGM_TABLES::MYSQL_REPLICATION_HOSTGROUPS]);
+		
+		char* error = NULL;
+		int cols = 0;
+		int affected_rows = 0;
+		SQLite3_result* resultset = NULL;
+
+		const char* query = "SELECT DISTINCT hostname, port, '1' is_writer, status, reader_hostgroup, writer_hostgroup, mem_pointer FROM mysql_replication_hostgroups JOIN mysql_servers ON hostgroup_id=writer_hostgroup \
+							 UNION \
+							 SELECT DISTINCT hostname, port, '0' is_writer, status, reader_hostgroup, writer_hostgroup, mem_pointer FROM mysql_replication_hostgroups JOIN mysql_servers ON hostgroup_id=reader_hostgroup \
+							 ORDER BY hostname, port";
+		
+		mydb->execute_statement(query, &error, &cols, &affected_rows, &resultset);
+
+		hostgroup_server_mapping.clear();
```

Comment:
After discussion, this servers recreation will be now the default behavior. This is, table `mysql_servers` by `Admin` is now the source of truth used for servers creation by `Hostgroup_Manager`. This means, that any servers created by `Monitoring` actions will be destroyed if they doesn't match the configuration supplied by the user at `mysql_servers` table, for later being moved/copied to the proper hostgroups when monitoring actions takes place.

This invalidates the requirement of the structure to persist in memory mentioned in the previous comment, because the only important thing is now that monitoring actions always converge to the same final configuration, taking into account this potential recreation.

In other words, this removes the previous behavior of performing an implicit `UPDATE` over the `runtime_mysql_servers` from `Admin` (user config). This behavior may now be optional through an additional variable of the `mysql-monitor_*` kind.